runelite-client/src/main/java/net/runelite/client/plugins/chatnotifications/ChatNotificationsPlugin.java

```diff
@@ -76,6 +79,9 @@
 	private String usernameReplacer = "";
 	private Pattern highlightMatcher = null;
 
+	// Private message cache used to avoid duplicate notifications from ChatHistory.
```

Comment:
I don't understand the purpose of this. What problem is this solving?

Reply:
Whenever you world hop and have ChatHistory enabled, you will be notified of messages from the past that you may have already received notifications about.